common/djangoapps/student/tests/test_tasks.py

```diff
@@ -0,0 +1,54 @@
+"""
+Tests for the Sending activation email celery tasks
+"""
+
+import mock
+
+from django.test import TestCase
+from django.conf import settings
+from student.tasks import send_activation_email
+from boto.exception import NoAuthHandlerFound
+
+from lms.djangoapps.courseware.tests.factories import UserFactory
+
+
+class SendActivationEmailTestCase(TestCase):
+    """
+    Test for send activation email to user
+    """
+    def setUp(self):
+        """ Setup components used by each test."""
+        super(SendActivationEmailTestCase, self).setUp()
+        self.student = UserFactory()
+
+    @mock.patch('time.sleep', mock.Mock(return_value=None))
+    @mock.patch('student.tasks.log')
+    @mock.patch('django.core.mail.send_mail', mock.Mock(side_effect=NoAuthHandlerFound))
+    def test_send_email(self, mock_log):
+        """
+        Tests retries when the activation email doesn't send
+        """
+        from_address = 'task_testing@edX.com'
+        email_max_attempts = settings.RETRY_ACTIVATION_EMAIL_MAX_ATTEMPTS + 1
```

Comment:
Also, don't add the `+ 1` here, do it in the for loop. The implementation of which attempt it is should be separated from the number of attempts you intent to try